core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1225,33 +1228,51 @@ private void updateMediaSession(final PlayerStatus playerStatus) {
+                getString(R.string.fast_forward_label),
+                R.drawable.ic_notification_fast_forward
+        );
+        WearMediaSession.addWearExtrasToAction(fastForwardBuilder);
+        sessionState.addCustomAction(fastForwardBuilder.build());
+
+        WearMediaSession.mediaSessionSetExtraForWear(mediaSession);
+
+        // Workaround bug in Android Auto custom actions described at https://issuetracker.google.com/issues/207389461 by checking for API 31+
+        // as well.
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S || currentModeType == Configuration.UI_MODE_TYPE_CAR) {
+            // On Android Auto, custom actions are added in the following order around the play button, if no default
+            // actions are present: Near left, near right, far left, far right, additional actions panel
             sessionState.addCustomAction(
-                new PlaybackStateCompat.CustomAction.Builder(
-                        CUSTOM_ACTION_REWIND,
-                        getString(R.string.rewind_label), R.drawable.ic_notification_fast_rewind)
-                        .build());
+                    new PlaybackStateCompat.CustomAction.Builder(
+                            CUSTOM_ACTION_CHANGE_PLAYBACK_SPEED,
+                            getString(R.string.playback_speed),
+                            R.drawable.ic_notification_playback_speed
+                    ).build()
+            );
             sessionState.addCustomAction(
-                new PlaybackStateCompat.CustomAction.Builder(
-                        CUSTOM_ACTION_FAST_FORWARD,
-                        getString(R.string.fast_forward_label), R.drawable.ic_notification_fast_forward)
-                        .build());
+                    new PlaybackStateCompat.CustomAction.Builder(
+                            CUSTOM_ACTION_SKIP_TO_NEXT,
+                            getString(R.string.skip_episode_label),
+                            R.drawable.ic_notification_skip
+                    ).build()
+            );
         } else {
-            // This would give the PIP of videos a play button
-            capabilities = capabilities | PlaybackStateCompat.ACTION_PLAY;
-            if (uiModeManager.getCurrentModeType() == Configuration.UI_MODE_TYPE_WATCH) {
-                WearMediaSession.sessionStateAddActionForWear(sessionState,
-                        CUSTOM_ACTION_REWIND,
-                        getString(R.string.rewind_label),
-                        android.R.drawable.ic_media_rew);
-                WearMediaSession.sessionStateAddActionForWear(sessionState,
-                        CUSTOM_ACTION_FAST_FORWARD,
-                        getString(R.string.fast_forward_label),
-                        android.R.drawable.ic_media_ff);
-                WearMediaSession.mediaSessionSetExtraForWear(mediaSession);
-            }
+            // Prevent next button from showing up unless we aren't showing the custom skip action
```

Comment:
So the custom skip action is needed because of the order of the buttons?

Reply:
Yes. In my testing, there was no way to move the default skip action from its default position just to the right of the play button to any other position without removing it and creating a custom skip action instead.